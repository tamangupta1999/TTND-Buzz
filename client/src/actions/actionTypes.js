//LOGIN
export const START_LOGIN_PROCESS = 'START_LOGIN_PROCESS';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILED = 'LOGIN_FAILED';

//LOGOUT
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'LOGOUT_FAILED';

//UI CONFIG
export const UI_CONFIG_FETCH_SUCCESS = 'UI_CONFIG_FETCH_SUCCESS';
export const UI_CONFIG_FETCH_FAILED = 'FAILED';

//POST BUZZ
export const BUZZ_POST_START = 'BUZZ_POST_START';
export const BUZZ_POST_SUCCESS = 'BUZZ_POST_SUCCESS'
export const BUZZ_POST_FAILED = 'BUZZ_POST_FAILED';

//FETCH ALL BUZZ 
export const BUZZ_FETCH_START = 'BUZZ_FETCH_START';
export const BUZZ_FETCH_SUCCESS = 'BUZZ_FETCH_SUCCESS'
export const BUZZ_FETCH_FAILED = 'BUZZ_FETCH_FAILED';

//FETCH My BUZZ 
export const MY_BUZZ_FETCH_START = 'MY_BUZZ_FETCH_START';
export const MY_BUZZ_FETCH_SUCCESS = 'MY_BUZZ_FETCH_SUCCESS'
export const MY_BUZZ_FETCH_FAILED = 'MY_BUZZ_FETCH_FAILED';

//DELETE BUZZ BY ID
export const DELETE_BUZZ_START = 'DELETE_BUZZ_START';
export const DELETE_BUZZ_SUCCESS = 'DELETE_BUZZ_SUCCESS';
export const DELETE_BUZZ_FAILED = 'DELETE_BUZZ_FAILED';

//UPDATE BUZZ
export const BUZZ_UPDATE_START = 'BUZZ_UPDATE_START'
export const BUZZ_UPDATE_SUCCESS = 'BUZZ_UPDATE_SUCCESS';
export const BUZZ_UPDATE_FAILED = 'BUZZ_UPDATE_FAILED';

//MARK LIKE ON BUZZ
export const LIKE_MARK_SUCCESS = 'LIKE_MARK_SUCCESS';
export const LIKE_MARK_FAILED = 'LIKE_MARK_FAILED';

//MARK DISLIKE ON BUZZ
export const DISLIKE_MARK_SUCCESS = 'DISLIKE_MARK_SUCCESS';
export const DISLIKE_MARK_FAILED = 'DISLIKE_MARK_FAILED';

//POST COMPLAINT
export const COMPLAINT_POST_START = 'COMPLAINT_POST_START';
export const COMPLAINT_POST_SUCCESS = 'COMPLAINT_POST_SUCCESS'
export const COMPLAINT_POST_FAILED = 'COMPLAINT_POST_FAILED';

//FETCH COMPLAINTS
export const USER_COMPLAINT_FETCH_START = 'USER_COMPLAINT_FETCH_START';
export const USER_COMPLAINT_FETCH_SUCCESS = 'USER_COMPLAINT_FETCH_SUCCESS'
export const USER_COMPLAINT_FETCH_FAILED = 'USER_COMPLAINT_FETCH_FAILED';

//FETCH COMPLAINT
export const COMPLAINT_FETCH_START = 'COMPLAINT_FETCH_START';
export const COMPLAINT_FETCH_SUCCESS = 'COMPLAINT_FETCH_SUCCESS'
export const COMPLAINT_FETCH_FAILED = 'COMPLAINT_FETCH_FAILED';

//UPDATE COMPLAINT
export const COMPLAINT_UPDATE_SUCCESS = 'COMPLAINT_UPDATE_SUCCESS'
export const COMPLAINT_UPDATE_FAILED = 'COMPLAINT_UPDATE_FAILED';

//ADD COMMENT ON COMPLAINT
export const COMMENT_ADD_SUCCESS = 'COMMENT_ADD_SUCCESS'
export const COMMENT_ADD_FAILED = 'COMMENT_ADD_FAILED';

//DELETE COMPLAINT
export const DELETE_COMPLAINT_SUCCESS = 'DELETE_COMPLAINT_SUCCESS'
export const DELETE_COMPLAINT_FAILED = 'DELETE_COMPLAINT_FAILED';

//FETCH COMPLAINTS
export const ALL_COMPLAINT_FETCH_START = 'ALL_COMPLAINT_FETCH_START';
export const ALL_COMPLAINT_FETCH_SUCCESS = 'ALL_COMPLAINT_FETCH_SUCCESS'
export const ALL_COMPLAINT_FETCH_FAILED = 'ALL_COMPLAINT_FETCH_FAILED';

//UPDATE STATUS ON COMPLAINT
export const UPDATE_STATUS_COMPLAINT_SUCCESS = 'UPDATE_STATUS_COMPLAINT_SUCCESS';
export const UPDATE_STATUS_COMPLAINT_FAILED = 'UPDATE_STATUS_COMPLAINT_FAILED';

//UPDATE CURRENT PAGE
export const UPDATE_CURRENT_PAGE = 'UPDATE_CURRENT_PAGE';

