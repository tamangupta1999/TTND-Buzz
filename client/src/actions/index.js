export {
    autheticateUser,
    logout,
    fetchUiConfig
}
    from './login';

export {
    buzzPost,
    fetchBuzz,
    deleteBuzz,
    markLike,
    markDislike,
    fetchMyBuzz,
    updateBuzz
}
    from './buzz';

export {
    complaintPost,
    fetchUserComplaints,
    fetchComplaint,
    addComment,
    deleteComplaint
}
    from './complaint';

export {
    updateComplaint,
    updateStatus,
    fetchAllComplaints,
    updateCurrentPage
}
    from './adminComplaint';