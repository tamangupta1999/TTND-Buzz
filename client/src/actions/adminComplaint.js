import axios from 'axios';
import * as actionTypes from './actionTypes';
import { toast } from 'react-toastify';


//Fetch User Complaints
const startFetchingAllComplaints = () => {
    return {
        type: actionTypes.ALL_COMPLAINT_FETCH_START
    }
}

export const fetchingAllComplaintsSuccess = (complaints, totalPages) => {
    return {
        type: actionTypes.ALL_COMPLAINT_FETCH_SUCCESS,
        payload: {
            complaints: complaints,
            totalPages: totalPages
        }
    }
}

export const fetchingAllComplaintsFailed = (error) => {
    return {
        type: actionTypes.ALL_COMPLAINT_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchAllComplaints = (pageNumber, size) => {
    return dispatch => {
        dispatch(startFetchingAllComplaints());
        axios.get(`/complaint/all/${pageNumber}/${size}`)
            .then(response => {
                let allComplaints = response.data.data;
                let totalPages = response.data.pages;
                dispatch(fetchingAllComplaintsSuccess(allComplaints, totalPages));
            })
            .catch(error => {
                dispatch(fetchingAllComplaintsFailed(error.message))
            })
    }
}

// Update Some Fields Of Complaints
const updateComplaintSuccess = () => {
    return {
        type: actionTypes.COMPLAINT_UPDATE_SUCCESS
    }
}
const updateComplaintFailed = (error) => {
    return {
        type: actionTypes.COMPLAINT_UPDATE_FAILED,
        payload: {
            error: error
        }
    }
}

export const updateComplaint = (complaintId, data) => {
    return dispatch => {
        axios.patch(`/complaint/update/${complaintId}`, {
            data: data
        }).then(response => {
            toast.success(response.data.message)
            dispatch(updateComplaintSuccess())
        }).catch(error => {
            toast.error(error.message)
            updateComplaintFailed(error.message)
        })
    }
}

//Update Status On Complaint Handler
const updateStatusOnComplaintSuccess = (data) => {
    return {
        type: actionTypes.UPDATE_STATUS_COMPLAINT_SUCCESS,
        payload: {
            complaint: data
        }
    }
}
const updateStatusOnComplaintFailed = () => {
    return {
        type: actionTypes.UPDATE_STATUS_COMPLAINT_FAILED
    }
}

export const updateStatus = (complaintId, status) => {
    return dispatch => {
        axios.patch(`/complaint/update/status/${complaintId}`, {
            data: {
                status: status
            }
        }).then(response => {
            toast.success(response.data.message);
            let updatedStatusComplaint = response.data.data;
            dispatch(updateStatusOnComplaintSuccess(updatedStatusComplaint))
        }).catch(error => {
            toast.error(error.message);
            dispatch(updateStatusOnComplaintFailed(error.message))
        })
    }
}

//Update Current Page
export const updateCurrentPage = (currentPage) => {
    return {
        type: actionTypes.UPDATE_CURRENT_PAGE,
        payload: {
            currentPage: currentPage
        }
    }
}
