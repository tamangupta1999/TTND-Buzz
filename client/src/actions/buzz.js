import axios from 'axios';
import * as actionTypes from './actionTypes';
import { toast } from 'react-toastify';

//Buzz Post Handler
const buzzPostStart = () => {
    return {
        type: actionTypes.BUZZ_POST_START
    }
}

const buzzPostSuccess = (message) => {
    return {
        type: actionTypes.BUZZ_POST_SUCCESS,
        payload: {
            message: message
        }
    }
}

const buzzPostFailed = (error) => {
    return {
        type: actionTypes.BUZZ_POST_FAILED,
        payload: {
            error: error
        }
    }
}

export const buzzPost = (formdata) => {
    return dispatch => {
        dispatch(buzzPostStart());
        axios.post('/buzz/create', formdata)
            .then(response => {
                toast.success(response.data.message);
                let message = response.data.message;
                dispatch(buzzPostSuccess(message));
            })
            .catch(error => {
                toast.error(error.message)
                dispatch(buzzPostFailed(error.message))
            })
    }
}

//Get All Buzz 
const startFetchingBuzz = () => {
    return {
        type: actionTypes.BUZZ_FETCH_START
    }
}

const fetchingBuzzSuccess = (buzzs, totalPage) => {
    return {
        type: actionTypes.BUZZ_FETCH_SUCCESS,
        payload: {
            totalPage: totalPage,
            buzzs: buzzs
        }
    }
}

const fetchingBuzzFailed = (error) => {
    return {
        type: actionTypes.BUZZ_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchBuzz = (pageNumber, size) => {
    return dispatch => {
        dispatch(startFetchingBuzz());
        axios.get(`/buzz/all/${pageNumber}/${size}`)
            .then(response => {
                let allBuzz = response.data.data;
                let totalPage = response.data.pages;
                dispatch(fetchingBuzzSuccess(allBuzz, totalPage));
            })
            .catch(error => {
                toast.error(error.message);
                dispatch(fetchingBuzzFailed(error.message))
            })
    }
}

// Fetch My Buzz 
const startFetchingMyBuzz = () => {
    return {
        type: actionTypes.MY_BUZZ_FETCH_START
    }
}

const fetchingMyBuzzSuccess = (buzzs) => {
    return {
        type: actionTypes.MY_BUZZ_FETCH_SUCCESS,
        payload: {
            buzzs: buzzs
        }
    }
}

const fetchingMyBuzzFailed = (error) => {
    return {
        type: actionTypes.MY_BUZZ_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchMyBuzz = () => {
    return dispatch => {
        dispatch(startFetchingMyBuzz());
        axios.get('/buzz/myBuzz').then(response => {
            let myBuzz = response.data.data;
            dispatch(fetchingMyBuzzSuccess(myBuzz));
        })
            .catch(error => {
                toast.error(error.message);
                dispatch(fetchingMyBuzzFailed(error.message))
            })
    }
}

// Delete Buzz By Id
const deleteBuzzStart = () => {
    return {
        type: actionTypes.DELETE_BUZZ_START
    }
}

const deleteBuzzSuccess = (buzzId) => {
    return {
        type: actionTypes.DELETE_BUZZ_SUCCESS,
        payload: {
            buzzId: buzzId
        }
    }
}

const deleteBuzzFailed = (error) => {
    return {
        type: actionTypes.DELETE_BUZZ_FAILED,
        payload: {
            error: error
        }
    }
}

export const deleteBuzz = (buzzId) => {
    return dispatch => {
        dispatch(deleteBuzzStart())
        axios.delete(`buzz/${buzzId}`).then(response => {
            toast.success(response.data.message);
            dispatch(deleteBuzzSuccess(buzzId))
        }).catch(error => {
            toast.error(error.message)
            dispatch(deleteBuzzFailed(error.message))
        })
    }
}

// Update Buzz Data 
const updateBuzzStart = () => {
    return {
        type: actionTypes.BUZZ_UPDATE_START
    }
}
const updateBuzzSuccess = (buzz) => {
    return {
        type: actionTypes.BUZZ_UPDATE_SUCCESS,
        payload: {
            buzz: buzz
        }
    }
}

const updateBuzzFailed = (error) => {
    return {
        type: actionTypes.BUZZ_UPDATE_FAILED,
        payload: {
            error: error
        }
    }
}

export const updateBuzz = (buzzId, data) => {
    return dispatch => {
        dispatch(updateBuzzStart())
        axios.patch(`buzz/update/${buzzId}`, data)
            .then(response => {
                toast.success(response.data.message)
                dispatch(updateBuzzSuccess(response.data.data))
            }).catch(error => {
                toast.error(error.message)
                dispatch(updateBuzzFailed(error.message))
            })
    }
}

//Mark Like 
const markLikeSuccess = (buzz) => {
    return {
        type: actionTypes.LIKE_MARK_SUCCESS,
        payload: {
            buzz: buzz
        }
    }
}

const markLikeFailed = (error) => {
    return {
        type: actionTypes.LIKE_MARK_FAILED,
        payload: {
            error: error
        }
    }
}

export const markLike = (buzzId) => {
    return dispatch => {
        axios.post(`buzz/like/${buzzId}`).then(response => {
            dispatch(markLikeSuccess(response.data.data))
        })
            .catch(error => {
                toast.error(error.message)
                dispatch(markLikeFailed(error.message))
            })
    }
}

//Mark Dislike 
const markDislikeSuccess = (buzz) => {
    return {
        type: actionTypes.DISLIKE_MARK_SUCCESS,
        payload: {
            buzz: buzz
        }
    }
}

const marklDisikeFailed = (error) => {
    return {
        type: actionTypes.DISLIKE_MARK_FAILED,
        payload: {
            error: error
        }
    }
}

export const markDislike = (buzzId) => {
    return dispatch => {
        axios.post(`buzz/dislike/${buzzId}`).then(response => dispatch(markDislikeSuccess(response.data.data)))
            .catch(error => dispatch(marklDisikeFailed(error.message)))
    }
}
