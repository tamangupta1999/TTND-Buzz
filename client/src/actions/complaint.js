import axios from 'axios';
import * as actionTypes from './actionTypes';
import { toast } from 'react-toastify';


//COMPLAINT Post Handler
const complaintPostStart = () => {
    return {
        type: actionTypes.COMPLAINT_POST_START
    }
}

const complaintPostSuccess = (data) => {
    return {
        type: actionTypes.COMPLAINT_POST_SUCCESS,
        payload: {
            message: data
        }
    }
}

const complaintPostFailed = (error) => {
    return {
        type: actionTypes.COMPLAINT_POST_FAILED,
        payload: {
            error: error
        }
    }
}

export const complaintPost = (formdata) => {
    return dispatch => {
        dispatch(complaintPostStart());
        axios.post('/complaint/create', formdata)
            .then(response => {
                toast.success(response.data.message);
                let message = response.data.message;
                dispatch(complaintPostSuccess(message));
                dispatch(fetchUserComplaints());
            })
            .catch(error => {
                toast.error(error.message)
                dispatch(complaintPostFailed(error.message))
            })
    }
}

//Fetch User Complaints
const startFetchingUserComplaints = () => {
    return {
        type: actionTypes.USER_COMPLAINT_FETCH_START
    }
}

const fetchingUserComplaintsSuccess = (complaints) => {
    return {
        type: actionTypes.USER_COMPLAINT_FETCH_SUCCESS,
        payload: {
            complaints: complaints
        }
    }
}

const fetchingUserComplaintsFailed = (error) => {
    return {
        type: actionTypes.USER_COMPLAINT_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchUserComplaints = () => {
    return dispatch => {
        dispatch(startFetchingUserComplaints());
        axios.get('/complaint/userComplaint')
            .then(response => {
                let allComplaints = response.data.data;
                dispatch(fetchingUserComplaintsSuccess(allComplaints));
            })
            .catch(error => {
                toast.error(error.message);
                dispatch(fetchingUserComplaintsFailed(error.message))
            })
    }
}

//Fetch Single Complaint
const startFetchingComplaint = () => {
    return {
        type: actionTypes.COMPLAINT_FETCH_START
    }
}

const fetchingComplaintSuccess = (complaint) => {
    return {
        type: actionTypes.COMPLAINT_FETCH_SUCCESS,
        payload: {
            complaint: complaint
        }
    }
}

const fetchingComplaintFailed = (error) => {
    return {
        type: actionTypes.COMPLAINT_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchComplaint = (complaintId) => {
    return dispatch => {
        dispatch(startFetchingComplaint());
        axios.get(`/complaint/${complaintId}`).then(response => {
            let complaint = response.data.data;
            dispatch(fetchingComplaintSuccess(complaint));
        })
            .catch(error => {
                toast.error(error.message);
                dispatch(fetchingComplaintFailed(error.message))
            })
    }
}
//Delete Complaint Handler
const deleteComplaintSuccess = () => {
    return {
        type: actionTypes.DELETE_COMPLAINT_SUCCESS
    }
}
const deleteComplaintFailed = () => {
    return {
        type: actionTypes.DELETE_COMPLAINT_FAILED
    }
}

export const deleteComplaint = (complaintId) => {
    return dispatch => {
        axios.delete(`/complaint/${complaintId}`)
            .then(response => {
                toast.success(response.data.message)
                dispatch(deleteComplaintSuccess());
            })
            .catch(error => {
                toast.error(error.message);
                dispatch(deleteComplaintFailed())
            })
    }
}
//Add Comment On Complaint Handler
const commentAddSuccess = (data) => {
    return {
        type: actionTypes.COMMENT_ADD_SUCCESS,
        payload: {
            data: data
        }
    }
}
const commentAddFailed = (error) => {
    return {
        type: actionTypes.COMMENT_ADD_FAILED,
        payload: {
            error: error
        }
    }
}

export const addComment = (complaintId, data) => {
    return dispatch => {
        axios.patch(`http://localhost:3000/complaint/add/comment/${complaintId}`, {
            data: data
        }).then(response => {
            toast.success(response.data.message)
            let updatedData = response.data.data;
            dispatch(commentAddSuccess(updatedData))
        }).catch(error => {
            toast.error(error.message)
            commentAddFailed(error.message)
        })
    }
}