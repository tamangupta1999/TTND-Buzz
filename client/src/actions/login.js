import axios from 'axios';
import * as actionTypes from './actionTypes';
import { toast } from 'react-toastify';


//login process
const loginStart = () => {
    return {
        type: actionTypes.START_LOGIN_PROCESS
    }

}
const autheticationSuccess = (response) => {
    return {
        type: actionTypes.LOGIN_SUCCESS,
        payload: response
    }
}
const autheticationFailed = (error) => {
    return {
        type: actionTypes.LOGIN_FAILED,
        payload: error
    }
}

export const autheticateUser = () => {
    return dispatch => {
        dispatch(loginStart());
        axios.get('/currentUser')
            .then(response => {
                if (response.data.error) {
                    toast.error(response.data.message)
                    dispatch(autheticationFailed(response.data))
                }
                else {
                    dispatch(autheticationSuccess(response.data))
                }

            })
            .catch(error => dispatch(autheticationFailed(error)))

    }
}

//Logout Process
const logoutSuccess = (message) => {
    console.log(message)
    return {
        type: actionTypes.LOGOUT_SUCCESS,
        payload: {
            message: message
        }
    }
}
const logoutFailed = (error) => {
    return {
        type: actionTypes.LOGOUT_FAILED,
        payload: {
            error: error
        }
    }
}

export const logout = () => {
    return dispatch => {
        axios({
            method: 'post',
            url: '/logout',
        }).then(response => {
            toast.success(response.data.message)
            dispatch(logoutSuccess(response.data.message))
        })
            .catch(error => {
                toast.error('Some Error Occured');
                dispatch(logoutFailed(error.message))
            })
    }
}

// Ui Config

const uiConfigFetchSuccess = (data) => {
    return {
        type: actionTypes.UI_CONFIG_FETCH_SUCCESS,
        payload: {
            data: data
        }
    }
}

const uiConfigFetchFailed = (error) => {
    return {
        type: actionTypes.UI_CONFIG_FETCH_FAILED,
        payload: {
            error: error
        }
    }
}

export const fetchUiConfig = () => {
    return dispatch => {
        axios.get('/uiconfig')
            .then(response => dispatch(uiConfigFetchSuccess(response.data.data)))
            .catch(error => dispatch(uiConfigFetchFailed(error.message)))
    }
}