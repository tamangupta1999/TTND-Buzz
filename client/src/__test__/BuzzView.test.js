import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import BuzzView from './../components/Buzz/BuzzViewComponent/BuzzViewComponent';
import { markLike } from './../actions';


const defaultProps = {
    email: "dummy@email.com",
    clickLike: jest.fn(() => console.log('like')),
    clickDislike: jest.fn(() => console.log('Dislike')),
    buzz: {
        createdOn: "2020-06-24T09:07:43.861Z",
        buzzImage: [],
        likes: ["dummy@email.com"],
        dislikes: [],
        createdBy: "dummy@email.com",
        category: "ACITVITY",
        buzzMessage: `Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.`,
        buzzId: "zi_rZIz1T",
    }

}

afterEach(cleanup);

describe('<BuzzViewComponent />', () => {
    // it('testing a <BuzzView /> Snapshot', () => {
    //     const { asFragment } = render(<BuzzView {...defaultProps} />);
    //     expect(asFragment(<BuzzView {...defaultProps} />)).toMatchSnapshot();
    // });

    it('should Created By is same as Default Props createdBy', () => {
        const { getByTestId } = render(<BuzzView {...defaultProps} />);
        expect(getByTestId('createdBy').textContent).toBe(defaultProps.buzz.createdBy)
    });

    it('should like button is working', () => {
        const { getByTestId } = render(<BuzzView {...defaultProps} />);

        let fired = fireEvent.click(getByTestId('like'));
        expect(fired).toBeTruthy();
    })
    it('verify class on like', () => {
        const wrapper = render(<BuzzView {...defaultProps} />).getByTestId('like');
        expect(wrapper.className).toEqual('Like AfterLike');
    })

    it('should buzzMessage is changed if length greater then 500 words', () => {
        const wrapper = render(<BuzzView {...defaultProps} />).getByTestId('buzzMessage');
        expect(wrapper.textContent).toEqual(`@taman.gupta${defaultProps.buzz.buzzMessage.substring(0, 500)}...`);
    })

    it('should like count is 1', async () => {
        await markLike('zi_rZIz1T');
        const wrapper = shallow(<BuzzView {...defaultProps} />);
        expect(wrapper.find('#like-count').text()).toEqual("1");
    });

    it('should dislike count is 0 ', () => {
        const wrapper = shallow(<BuzzView {...defaultProps} />);
        expect(wrapper.find('#dislike-count').text()).toEqual("0");
    });

})

