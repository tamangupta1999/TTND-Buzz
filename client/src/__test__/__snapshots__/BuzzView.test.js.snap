// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuzzViewComponent /> testing a <BuzzView /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="BuzzViewComponent"
  >
    <p
      class="ShowDate"
    >
      24
      <br />
      06
    </p>
    <div
      class="BuzzContentCard"
    >
      <div
        class="BuzzContentHeader"
      >
        <div
          class="BuzzContentHeader"
        >
          <p
            class="BuzzCreatedBy"
            data-testid="createdBy"
          >
            dummy@email.com
          </p>
          <p
            class="BuzzCreatedOn"
          >
            ~21 days ago
          </p>
        </div>
      </div>
      <div
        class="BuzzCard"
      >
        <p
          class="BuzzMessage"
          data-testid="buzzMessage"
        >
          <span
            style="color: grey; display: inline-block; text-decoration-line: underline; margin-bottom: 10px;"
          >
            @taman.gupta
          </span>
          <br />
          Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. He he in forfeited furniture sweetness he arranging. Me tedious so to behaved written account ferrars moments. Too objection for elsewhere her preferred allowance her. Marianne shutters mr steepest to me.
    Bringing unlocked me an striking ye perceive. Mr by wound hours oh happy. Me in resolution pianoforte continuing we. Most my no spot felt by no. H...
        </p>
      </div>
      <div
        class="StatusCard"
      >
        <p
          class="Like AfterLike"
          data-testid="like"
        >
          <i
            class="fas fa-thumbs-up"
          />
          1
        </p>
        <p
          class="Dislike"
        >
          <i
            class="fas fa-thumbs-down"
          />
          0
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
